gdal/frmts/gtiff/geotiff.cpp

```diff
@@ -17826,6 +17826,8 @@ GTiffDataset::CreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
         }
 #endif
     }
+
+#if defined(HAVE_LIBJPEG) || defined(JPEG_DIRECT_COPY)
```

Comment:
This is incorrect. Only the "bTryCopy &&" part should be protected.
So
```
    else if( 
 #if defined(HAVE_LIBJPEG) || defined(JPEG_DIRECT_COPY)
             bTryCopy &&
#endif
             eErr == CE_None )
{
```

Reply:
Thanks, fixed.